test/unit/specs/core/edgeNetwork/sendEdgeNetworkRequestFactory.spec.js

```diff
@@ -395,31 +395,36 @@ describe("sendEdgeNetworkRequestFactory", () => {
 
   it("returns the object from lifecycle::onBeforeRequest", () => {
     lifecycle.onBeforeRequest.and.callFake(({ onResponse }) => {
-      onResponse(() => [{ a: 1 }]);
+      onResponse(() => ({ a: 1 }));
+      onResponse(() => ({ b: 2 }));
+      onResponse(() => ({ c: 3 }));
+      onResponse(() => undefined);
       return Promise.resolve();
     });
 
     return expectAsync(
       sendEdgeNetworkRequest({ payload, action })
-    ).toBeResolvedTo({ a: 1 });
+    ).toBeResolvedTo({ a: 1, b: 2, c: 3 });
   });
 
   it("returns the object from lifecycle::onResponse", () => {
-    lifecycle.onResponse.and.returnValue([{ a: 1 }]);
+    lifecycle.onResponse.and.returnValue(
+      Promise.resolve([{ c: 2 }, { h: 9 }, undefined])
+    );
 
     return expectAsync(
       sendEdgeNetworkRequest({ payload, action })
-    ).toBeResolvedTo({ a: 1 });
+    ).toBeResolvedTo({ c: 2, h: 9 });
```

Comment:
The `returns the object from lifecycle::onBeforeRequest` and `returns the object from lifecycle::onResponse` tests seem like they're both already covered in the `returns the merged object from  lifecycle::onBeforeRequest & lifecycle::onResponse` test and are therefore redundant. I would prefer keeping `returns the merged object from  lifecycle::onBeforeRequest & lifecycle::onResponse` (because it tests that objects from both `lifecycle::onBeforeRequest` and `lifecycle::onResponse` are properly merged together) and getting rid of the other two.

We don't have any test that tests that the return values from `runOnResponseCallbacks` get merged into the final return value. I would recommend augmenting the `returns the merged object from  lifecycle::onBeforeRequest & lifecycle::onResponse` test to include that.

Reply:
ok, I'll add that. thank you